### ISSUE TITLE
Question: Radial Chart data level override 

### ISSUE THREAD

**Issue by roderickhsiao** [uber/react-vis#234]
From the custom radial chart example seems like we can pass data level override, 
for instance we tried to do on hover on section to have focus style (change `opacity`, `radius` and `innerRadius`), but the chart seemly doesnt work that way.

1. Opacity: 
As long as we has opacity in the data, every arcs render as opacity `0.1`
```js
const data = [
      {angle: 2, title: '1', opacity: 1},
      {angle: 6, title: '2', opacity: 1},
      {angle: 2, title: '3', opacity: 1},
      {angle: 3, title: '4', opacity: 1},
      {angle: 1, title: '5', opacity: 1}
    ];
]
```
==> all ars render opacity 0.1

If data looks like
```js
const data = [
      {angle: 2, title: '1', opacity: .5},
      {angle: 6, title: '2', opacity: .5},
      {angle: 2, title: '3', opacity: .5},
      {angle: 3, title: '4', opacity: .5},
      {angle: 1, title: '5', opacity: 1}
    ];
]
```
==> Only the one with `opacity:1` will be correct, other render as opacity `0.1`

2. Data level `innerRadius` and `radius` doesnt seem to work if we had component level props passed in.

**Comment by mcnuttandrew** [comment]
Hey @roderickhsiao 

I just merged into a pr which i think should address the issue. I think what was causing it, was the automatic generation of domains. I'm planning on doing another minor later today once sankey etc lands, would you mind checking to see if this is resolved once that happens?

**Comment by roderickhsiao** [comment]
sure thing thanks!

**Comment by roderickhsiao** [comment]
@mcnuttandrew I tried the version 0.10.2,
the first case works, but the second one

```js
const data = [
      {angle: 2, title: '1', opacity: .5},
      {angle: 6, title: '2', opacity: .5},
      {angle: 2, title: '3', opacity: .5},
      {angle: 3, title: '4', opacity: .5},
      {angle: 1, title: '5', opacity: 1}
    ];
]
```
the rest still showing opacity: 0.1

**Comment by roderickhsiao** [comment]
also tried changing radius and innerRadius on over section doesnt' seem to work

**Comment by roderickhsiao** [comment]
Found one issue,

if the data looks like
```js
const data = [
      {angle: 2, title: '1', opacity: .5},
      {angle: 6, title: '2'},
      {angle: 2, title: '3'},
      {angle: 3, title: '4'},
      {angle: 1, title: '5'}
    ];
]
```

it will return `null` from `_getAttributeFunctor`, and react will throw error on setting nan to opacity

**Comment by roderickhsiao** [comment]
Simple test case to show
```js
import React from 'react';
import {
	RadialChart,
  makeWidthFlexible
} from 'react-vis';

const FlexibleRadialChart = makeWidthFlexible(RadialChart);

const PROPS_BY_TYPE = {
	donut: {
		innerRadius: 100,
    radius: 140
	},
	'donut-focus': {
		innerRadius: 80,
    radius: 160
	}
};

const getChartData = (count) => {
	var data = [];
	for (var i = 0; i < count; i++) {
		data.push({
			angle: Math.floor((Math.random() * 10) + 1),
			title: `data-${i}`
		});
	}
	return data;
}

const isNil = (data) => (typeof data === undefined) || (data === null);
/**
 * Generic Line Chart Component
 */
class BranchRadialChart extends React.Component {
	static displayName = 'BranchRadialChart'

	static defaultProps = {
		height: 300,
		type: 'donut'
	}

	constructor(props) {
		super(props);
		this.state = {
			active: null,
			focus: null
		};
		this.data = getChartData(10);
	}

	getDataWithStyle = (data) => {
		const { focus, active } = this.state;
		const { type } = this.props;
		const focusProps = PROPS_BY_TYPE[`${type}-focus`] || {};

		return data.map((d) => {
			const isFocus = isNil(focus) ? true : (focus === d.title);
			var props;
			if (isFocus) {
				props = Object.assign({}, focusProps, { opacity: 1 });
			}
			else {
				props = { opacity: .5 };
			}

			return Object.assign({}, props, d);
		});
	}

	render() {
		const {
			type,
			height,
			...otherProps
		} = this.props;
		const propsByType = PROPS_BY_TYPE[type] || {};

		return (
			<div className="chart-wrapper-container">
				<FlexibleRadialChart
					data={this.getDataWithStyle(this.data)}
					height={height}
					className="chart-wrapper"
					onSectionMouseOver={(v) => this.setState({ focus: v.title })}
					onSectionMouseOut={(v) => this.setState({ focus: null })}
					{...otherProps}
					{...propsByType}
				>
				</FlexibleRadialChart>
			</div>
		);
	}
}

export default BranchRadialChart;
```

**Comment by mcnuttandrew** [comment]
Hey @roderickhsiao 

So it turns out there were a few things driving this bug. First among them was innerRadius was getting marked into a categorical scale, which (#377) over-rides. I reasonably certain that the new version of the radial chart should address this issue (available on 2.0.0-beta), but you should double check.

Otherwise i think this can be closed.

**Comment by roderickhsiao** [comment]
👍  https://roderickhsiao.github.io/react-vis-random/?selectedStory=#234

Verified
